File: 프로그래머스/1/135808. 과일 장수/과일 장수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, int m, vector<int> score) {
    int answer = 0;
    int currentAppleCount = 0;
    int minScore = k;
    
    
    sort(score.begin(), score.end(),greater<int>());
    
    for(auto itr = score.begin(); itr != score.end(); itr++)
    {
        ++currentAppleCount;
        minScore = min(minScore,*itr);
        if(currentAppleCount >= m)
        {
            answer += currentAppleCount * minScore;
            currentAppleCount = 0;
            minScore = k;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.cs
using System;

public class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        
        Array.Sort(score,(x,y)=> x-y);
        
        //Array.ForEach(score, x => Console.WriteLine(x) );
                   
        for(int i=score.Length-1;i>=m-1;i-=m)
        {
            answer += score[i-m+1] * m;
        }
        return answer;
    }
}